compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -344,62 +344,61 @@ private Map<String, PackageManifest.Platform> getPlatforms(TopLevelNode platform
             if (platformNode.kind() == TomlType.TABLE) {
                 TomlTableNode platformTable = (TomlTableNode) platformNode;
                 Set<String> platformCodes = platformTable.entries().keySet();
-                if (platformCodes.size() != 1) {
-                    return platforms;
-                }
-
-                String platformCode = platformCodes.stream().findFirst().get();
-                TopLevelNode platformCodeNode = platformTable.entries().get(platformCode);
-
-                if (platformCodeNode.kind() == TomlType.TABLE) {
-                    TomlTableNode platformCodeTable = (TomlTableNode) platformCodeNode;
-                    // Get graalvmCompatible value
-                    TopLevelNode graalvmCompatibleNode = platformCodeTable.entries().get(GRAALVM_COMPATIBLE);
-                    if (graalvmCompatibleNode != null) {
-                        if (graalvmCompatibleNode.kind() == TomlType.NONE) {
-                            return platforms;
-                        }
-                        PackageManifest.Platform newPlatform = getGraalvmCompatibilityPlatform(graalvmCompatibleNode);
-                        if (newPlatform != null) {
-                            if (platforms.get(platformCode) != null) {
-                                newPlatform = new PackageManifest.Platform(platforms.get(platformCode).dependencies(),
-                                        platforms.get(platformCode).repositories(),
-                                        newPlatform.graalvmCompatible());
+                for (String platformCode : platformCodes) {
+                    TopLevelNode platformCodeNode = platformTable.entries().get(platformCode);
+
+                    if (platformCodeNode.kind() == TomlType.TABLE) {
+                        TomlTableNode platformCodeTable = (TomlTableNode) platformCodeNode;
+                        // Get graalvmCompatible value
+                        TopLevelNode graalvmCompatibleNode = platformCodeTable.entries().get(GRAALVM_COMPATIBLE);
+                        if (graalvmCompatibleNode != null) {
+                            if (graalvmCompatibleNode.kind() == TomlType.NONE) {
+                                return platforms;
```

Comment:
This would introduce a bug when we have two platforms and only the second platform has the graalvm compatibility property. Shall we test and fix this?

Reply:
Tested this and did not find any issues